iris/tests/test_ingestion_phase_cancellation.py

```
@@ -0,0 +1,606 @@
+        patch(
+            "iris.pipeline.shared.transcription.whisper_client._audio_duration",
+            return_value=1.0,
+        ),
+        patch(
+            "iris.pipeline.shared.transcription.whisper_client.requests.post",
+            side_effect=post,
+        ),
+    ):
+        with pytest.raises(RuntimeError):
+            client.transcribe(
+                "/tmp/audio.mp3", lecture_unit_id=3, on_chunk_complete=fail_run
```

Ruff (0.16.1)

[error] Probable insecure usage of temporary file or directory: "/tmp/audio.mp3"

(S108)